core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -526,7 +526,10 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener) throws Sql
                                     if (prevBranch != BRANCH_GEOHASH) {
                                         // validate type
                                         final short columnTypeTag = ColumnType.tagOf(node.token);
-                                        if (((columnTypeTag < ColumnType.BOOLEAN || columnTypeTag > ColumnType.LONG256) && !asPoppedNull) ||
+                                        //todo: this looks weird, I am not entirely sure what is going on here
```

Comment:
this is validation of `CAST(x as TYPE)` operator. It checks that `TYPE` is a supported one